web/src/settings_notifications.ts

```diff
@@ -137,10 +154,13 @@ function stream_notification_setting_changed(e) {
     }
 
     const $status_element = $(e.target).closest(".subsection-parent").find(".alert-notification");
-    const setting = e.target.name;
+    assert(e.target instanceof HTMLInputElement);
```

Comment:
This function only uses `e` for its `target`, so it could instead be passed the `HTMLInputElement` instead of the `JQuery.ChangeEvent`. The selector is an event handler for the selector `"input, select"` and `<input>` has no children, so there’s no difference here between `.target` and `.currentTarget`, a.k.a. the handler’s `this`, which could be typed `HTMLInputElement | HTMLSelectElement`. So we’d still need to check, but it’d make sense what we’re checking for.

(In general, I think almost all of our uses of `.target` should be `.currentTarget`/`this`, which has clearer semantics and less type confusion.)

Reply:
That makes sense. Added a prep commit for this, thanks!